File: src/dataset/javascript/matrix_exercism-elenamountz/claude/ClaudeSonnet4_matrix_v4.js
export class Matrix {
  constructor(str) {
    this.str = str;
    this._parseMatrix();
  }

  _parseMatrix() {
    const lines = this.str.split('\n');
    const numRows = lines.length;
    const numCols = lines[0] ? lines[0].split(' ').length : 0;
    
    this.rows = new Array(numRows);
    this.columns = new Array(numCols);
    
    for (let i = 0; i < numCols; i++) {
      this.columns[i] = new Array(numRows);
    }
    
    for (let i = 0; i < numRows; i++) {
      const values = lines[i].split(' ');
      this.rows[i] = new Array(numCols);
      
      for (let j = 0; j < numCols; j++) {
        const value = parseInt(values[j], 10);
        this.rows[i][j] = value;
        this.columns[j][i] = value;
      }
    }
  }

  getRows() {
    return this.rows;
  }

  getColumns() {
    return this.columns;
  }
}

File: src/dataset/Java/matrix_exercism-uzilan/claude/ClaudeSonnet4_Matrix_v4.java
import java.util.Arrays;

class Matrix {

    private final int[][] matrix;
    private final int rows;
    private final int columns;

    Matrix(String matrixAsString) {
        String[] lines = matrixAsString.split("\n");
        this.rows = lines.length;
        this.columns = lines[0].split(" ").length;
        this.matrix = new int[rows][columns];
        
        for (int i = 0; i < rows; i++) {
            String[] values = lines[i].split(" ");
            for (int j = 0; j < columns; j++) {
                matrix[i][j] = Integer.parseInt(values[j]);
            }
        }
    }

    int[] getRow(int rowNumber) {
        return Arrays.copyOf(matrix[rowNumber], columns);
    }

    int[] getColumn(int columnNumber) {
        int[] column = new int[rows];
        for (int i = 0; i < rows; i++) {
            column[i] = matrix[i][columnNumber];
        }
        return column;
    }

    int getRowsCount() {
        return rows;
    }

    int getColumnsCount() {
        return columns;
    }
}